### PR TITLE
python311Packages.netbox-bgp: init at 0.14.0

### PR DESCRIPTION
**Pull request by felbinger**
## Description of changes

NetBox plugin for BGP related objects documentation
https://github.com/netbox-community/netbox-bgp

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by minijackson** [comment]
@felbinger I think in terms of compatibility, you should choose whatever the `netbox` package is at, which is currently 3.7. Note that I've opened #331750 recently, so depending on which gets merged first, one of us will have to upgrade netbox-bgp.

File: pkgs/development/python-modules/netbox-bgp/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  netbox,
+}:
+
+buildPythonPackage rec {
+  pname = "netbox-bgp";
+  version = "0.11.1";
+  format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/netbox-bgp/default.nix:12]:
```suggestion
  pyproject = true;
```

**Review by SuperSandro2000 — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-2411.section.md
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2411.section.md:1]: Not sure if that warrants an entry

File: pkgs/development/python-modules/netbox-bgp/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+buildPythonPackage rec {
+  pname = "netbox-bgp";
+  version = "0.11.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "netbox-community";
+    repo = "netbox-bgp";
+    rev = "v${version}";
+    hash = "sha256-T/+/JxY9Oyf7e70yK8X/ZaENYbV0f0YmGYtaEmnvhgI";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/netbox-bgp/default.nix:19]:
```suggestion
    hash = "sha256-+F7Y+bmxVz/P1C8zG9Uev646zBKvlYOx/pENN1+4Dig=";
```
see ofborg logs

File: nixos/doc/manual/release-notes/rl-2411.section.md
Comment (felbinger) [reply nixos/doc/manual/release-notes/rl-2411.section.md:1]: Ok, removed.

File: pkgs/development/python-modules/netbox-bgp/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    repo = "netbox-bgp";
+    rev = "v${version}";
+    hash = "sha256-+F7Y+bmxVz/P1C8zG9Uev646zBKvlYOx/pENN1+4Dig=";
+  };
+
+  nativeBuildInputs = [ pytestCheckHook ];
+
+  build-system = [ setuptools ];
+
+  checkInputs = [ netbox ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/netbox-bgp/default.nix:26]:
```suggestion
  checkInputs = [ netbox ];

  pythonImportsCheck = [ "netbox_bgp" ];
```

File: pkgs/development/python-modules/netbox-bgp/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/netbox-bgp/default.nix:26]:
```suggestion
  nativeCheckInputs = [ netbox pytestCheckHook ];

  preFixup = ''
    export PYTHONPATH=${netbox}/opt/netbox/netbox:$PYTHONPATH
  '';

  dontUsePythonImportsCheck = python.pythonVersion != netbox.python.pythonVersion;
```

File: pkgs/development/python-modules/netbox-bgp/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "netbox-community";
+    repo = "netbox-bgp";
+    rev = "v${version}";
+    hash = "sha256-+F7Y+bmxVz/P1C8zG9Uev646zBKvlYOx/pENN1+4Dig=";
+  };
+
+  nativeBuildInputs = [ pytestCheckHook ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/netbox-bgp/default.nix:23]:
```suggestion
```

**Comment by minijackson** [comment]
The NetBox 4.1 upgrade was merged, so I think this plugin needs to be updated to v0.14.0

**Pull request merged**